packages/expo-updates/ios/EXUpdates/Update/EXUpdatesUpdate.m

```diff
@@ -58,16 +58,26 @@ + (instancetype)updateWithManifest:(NSDictionary *)manifest
                           response:(nullable NSURLResponse *)response
                             config:(EXUpdatesConfig *)config
                           database:(EXUpdatesDatabase *)database
+                             error:(NSError **)error
 {
-  if (config.usesLegacyManifest) {
+  NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+  NSDictionary *headerDictionary = [httpResponse allHeaderFields];
+  NSNumber *expoProtocolVersion = headerDictionary[@"expo-protocol-version"];
```

Comment:
let's wrap this in a `if ([response isKindOfClass:[NSHTTPURLResponse class]])` check